### PR TITLE
Add client support

### PR DESCRIPTION
**Pull request by ranj063**
This is needed for the SOF client implementation in https://github.com/thesofproject/linux/pull/1841

### CODE REVIEW

File: sof-mach-driver-defconfig [hunk 1]
```diff
@@ -1,4 +1,5 @@
 CONFIG_SND_SOC_SOF_HDA_LINK=y
+CONFIG_SND_SOC_SOF_HDA_DISABLE_DMI_L1=y
```
Comment (plbossart) [inline sof-mach-driver-defconfig:2]: is this really needed? I thought it's not recommended?

Comment (ranj063) [reply sof-mach-driver-defconfig:2]: I was surprised to see this as well. Let me send a PR to remove it

Comment (ranj063) [reply sof-mach-driver-defconfig:2]: oops sorry. its my messed up local branch!

File: sof-defconfig [hunk 2]
```diff
@@ -5,6 +5,7 @@ CONFIG_SND_SOC_SOF_TOPLEVEL=y
 CONFIG_SND_SOC_SOF_PCI=m
 CONFIG_SND_SOC_SOF_ACPI=m
 CONFIG_SND_SOC_SOF_OF=m
+CONFIG_SND_SOC_SOF_CLIENT=y
```
Comment (plbossart) [inline sof-defconfig:8]: is this y for boolean or can it be m? That will impact how the code is compiled.

Comment (ranj063) [reply sof-defconfig:8]: Its a boolean. so it cant be m, right?

**Pull request merged**